### PR TITLE
feat: honor unique keys in parent table for weak fks

### PR DESCRIPTION
**Pull request by TSchiefer**
a test is implemented as well.

closes https://github.com/cynkra/dm/issues/1131

### CODE REVIEW

File: R/examine-constraints.R [hunk 1]
```diff
@@ -90,11 +90,23 @@ print.dm_examine_constraints <- function(x, ...) {
...
 
 check_pk_constraints <- function(dm, progress = NA, top_level_fun = NULL) {
-  pks <- dm_get_all_pks_impl(dm)
+  pks <- dm_get_all_pks_impl(dm) %>%
+    mutate(kind = "PK") %>%
+    bind_rows(
+      dm_get_all_fks_impl(dm) %>%
+        select(table = parent_table, pk_col = parent_key_cols) %>%
+        distinct() %>%
+        mutate(kind = "UK")
```
Comment (krlmlr) [inline R/examine-constraints.R:107]: Extract `dm_get_all_uks_impl(dm)` ?

File: R/examine-constraints.R [hunk 2]
```diff
@@ -90,11 +90,23 @@ print.dm_examine_constraints <- function(x, ...) {
...
+    kind == "UK" ~ "unique key",
+    .default = "foreign key"
+  )
 }
 
 check_pk_constraints <- function(dm, progress = NA, top_level_fun = NULL) {
-  pks <- dm_get_all_pks_impl(dm)
+  pks <- dm_get_all_pks_impl(dm) %>%
+    mutate(kind = "PK") %>%
+    bind_rows(
```
Comment (krlmlr) [inline R/examine-constraints.R:103]: `bind_rows(PK = pks, UK = uks, .id = "kind")` ?

**Comment by krlmlr** [comment]
The snapshot update is extra weird, sorry about that.

**Review by krlmlr — APPROVED** [review]
Thanks, I expect more PRs coming in after the weird snapshot update. We could do one more iteration and merge, or leave it as is.

File: R/examine-constraints.R (hunk 2 above)
Comment (TSchiefer) [reply R/examine-constraints.R:103]: sadly, `.id = "kind"` doesn't take the info from the existing column "kind", but instead creates a new one with values "1" or "2". Can still make use of it though.

Comment (krlmlr) [reply R/examine-constraints.R]: Also with named arguments?

Comment (krlmlr) [reply R/examine-constraints.R]: "1" and "2" seem dangerous.

Comment (TSchiefer) [reply R/examine-constraints.R]: one step ahead of me, there, nice trick!

**Comment by TSchiefer** [comment]
BTW, had chosen "fix", because the issue was labeled as a bug.

**Comment by krlmlr** [comment]
Yeah, let's not be too strict with ourselves. It was *also* labeled an enhancement. 🙃

**Comment by TSchiefer** [comment]
So I think this PR might be OK now (confused about the check-failures though). Let me know if I can still do something here.

**Pull request merged**